client/src/components/EditProfile/AvailabilityTab/AvailabilityTab.tsx

```diff
@@ -0,0 +1,187 @@
+import editAvailability from '../../../helpers/APICalls/editAvailability';
+import fetchProfile from '../../../helpers/APICalls/fetchProfile';
+import { mockId } from '../../../mocks/mockId';
+
+const months = [
+  'January',
+  'February',
+  'March',
+  'April',
+  'May',
+  'June',
+  'July',
+  'August',
+  'September',
+  'October',
+  'November',
+  'December',
+];
+
+const days = [
+  {
+    day: 'Monday',
+    id: 1,
+  },
+  {
+    day: 'Tuesday',
+    id: 2,
+  },
+  {
+    day: 'Wednesday',
+    id: 3,
+  },
+  {
+    day: 'Thursday',
+    id: 4,
+  },
+  {
+    day: 'Friday',
+    id: 5,
+  },
+  {
+    day: 'Saturday',
+    id: 6,
+  },
+  {
+    day: 'Sunday',
+    id: 7,
+  },
+];
+
+const AvailabilityTab = (): JSX.Element => {
+  const classes = useStyles();
+  const [startDate, setStartDate] = useState(new Date(2019, 5, 17));
+  const [availability, setAvailability] = useState([
+    {
+      day: 'Monday',
+      startTime: 10,
+      endTime: 22,
+      date: new Date('2019-6-17'),
+    },
+    {
+      day: 'Tuesday',
+      startTime: 10,
+      endTime: 22,
+      date: new Date('2019-6-18'),
+    },
+    {
+      day: 'Wednesday',
+      startTime: 10,
+      endTime: 22,
+      date: new Date('2019-6-19'),
+    },
+    {
+      day: 'Thursday',
+      startTime: 10,
+      endTime: 22,
+      date: new Date('2019-6-20'),
+    },
+    {
+      day: 'Friday',
+      startTime: 10,
+      endTime: 22,
+      date: new Date('2019-6-21'),
+    },
+    {
+      day: 'Saturday',
+      startTime: 10,
+      endTime: 22,
+      date: new Date('2019-6-22'),
+    },
+    {
+      day: 'Sunday',
+      startTime: 10,
+      endTime: 22,
+      date: new Date('2019-6-23'),
+    },
+  ]);
+  const [updating, setUpdating] = useState(false);
+
+  const changeStartTime = (index: number, newTime: number, dayToUpdate: string) => {
+    setAvailability(
+      availability.map((day) => {
+        if (day.day === dayToUpdate) {
+          return {
+            day: day.day,
+            startTime: newTime,
+            endTime: day.endTime,
+            date: day.date,
+          };
+        } else {
+          return day;
+        }
+      }),
+    );
+    setUpdating(true);
+  };
+
+  const changeEndTime = (index: number, newTime: number, dayToUpdate: string) => {
+    setAvailability(
+      availability.map((day) => {
+        if (day.day === dayToUpdate) {
+          return {
+            day: day.day,
+            startTime: day.startTime,
+            endTime: newTime,
+            date: day.date,
+          };
+        } else {
+          return day;
+        }
+      }),
+    );
+    setUpdating(true);
+  };
+
+  useEffect(() => {
+    if (updating) {
+      editAvailability(mockId, availability).then((data) => console.log(data));
+      setUpdating(false);
+    }
+  }, [availability, updating]);
+
+  useEffect(() => {
+    fetchProfile(mockId).then((data) => setAvailability(data.success.profile.availableTime));
```

Comment:
I think you can have a chat with the team and move the profile state into a context (from which you can access the availability)

Reply:
How should we proceed with this issue?

Reply:
You can have a chat with @ababber and if we don't already have the logged in user's profile stored in a context, either one of you can create it 👍🏽